### ISSUE TITLE
Always prompts for password on Travis

### ISSUE THREAD

**Issue by RJPercival** [skeema/skeema#73]
When building on Travis with MySQL enabled, the skeema tool always prompts for a password, even though no password is required. I believe the reason is that Travis creates a file at ~/.my.cnf containing:
```ini
[client]
default-character-set = utf8
port = 3306
user = root
password =
socket = /var/run/mysqld/mysqld.sock
```

Skeema automatically detects and uses the password setting in this file, which it interprets as a request to prompt for a user. This is undesirable behaviour, since this config file is clearly indicating that no password is required. The absence of a password setting in this config file would indicate that prompting for a password is reasonable.

I am aware that I can pass --skip-my-cnf to skeema to avoid this problem. However, it took some time to discover that the ~/.my.cnf file was the cause, so this does not avoid other users running into the same problem and similarly having to spend time to diagnose it. I suggest either changing this behaviour or adding an FAQ entry to cover this.

**Comment by evanelias** [comment]
Thank you for the report. This seems to be a bug in how Skeema handles `password=`, i.e. when there's no value but the equals sign is still present.

The goal is for Skeema to match the behavior of the `mysql` client exactly, which it does in all other cases:
* `password` alone with no equals sign: prompt for password
* `password=''` i.e. explicit blank string: connect without a password
* no password option supplied at all: connect without a password

I'm surprised that Travis isn't using that last method -- the canonical way of telling `mysql` to connect without a password is to just omit the `password` option entirely. `password =` is super weird, I've never encountered that in my 16 years of using MySQL. Regardless, I'll get this fixed in Skeema soon, and will aim to post an update on this issue later today.

**Comment by evanelias** [comment]
On closer inspection, `password=''` i.e. explicit blank string doesn't currently work properly in Skeema either. Seems I broke this in a refactor long ago, or I misremembered handling that case... 

I'll need to give more thought to the best way to differentiate between these cases generically. String args that don't require a value are painful to parse in general, which is why `mysql` has special rules around spacing (see 5th bullet on https://dev.mysql.com/doc/refman/8.0/en/command-line-options.html)
